src/main/java/com/skhu/luxuryshop/product/controller/ProductController.java

```diff
@@ -0,0 +1,48 @@
+package com.skhu.luxuryshop.product.controller;
+
+import com.skhu.luxuryshop.product.dto.ProductRequestDto;
+import com.skhu.luxuryshop.product.service.ProductService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/products")
+@Slf4j
+public class ProductController {
+    private final ProductService productService;
+
+    @GetMapping
+    public ResponseEntity getProducts() {
+        return ResponseEntity.ok(productService.findAll());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity getProductById(@PathVariable Long id) {
+        return ResponseEntity.ok(productService.findById(id));
+    }
+
+    @PostMapping
+    public ResponseEntity<Long> registerProduct(@RequestBody @Valid ProductRequestDto productDto) {
+        productService.save(productDto);
+        return new ResponseEntity("상품등록 성공", HttpStatus.OK);
+    }
```

Comment:
POST 매핑으로 새로운 데이터를 저장했을 경우에는 200 OK도 가능하지만 
보통 201 created 상태 코드를 씁니다!
그리고 새로 생성 된 리소스를 반환받을 수 있는 uri를 header에 넣어줍니다!

응답한 데이터를 명확하게 표현하기 위한 ststus code 중에하나인 201 code도 공부해보길 추천해요!

productService.save를 조금 수정하고 이런식으로 수정하면 될 거예요.
```suggestion
        return ResponseEntity.created(URI.create("/" + savedProduct.getId())).build();
```